src/api/spec/controllers/staging/staging_projects_controller_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe Staging::StagingProjectsController, type: :controller, vcr: true do
+  render_views
+
+  let(:user) { create(:confirmed_user, login: 'permitted_user') }
+  let(:project) { user.home_project }
+  let(:staging_workflow) { create(:staging_workflow_with_staging_projects, project: project) }
+  let(:staging_project) { staging_workflow.staging_projects.first }
+  let(:source_project) { create(:project, name: 'source_project') }
+  let(:target_package) { create(:package, name: 'target_package', project: project) }
+  let(:source_package) { create(:package, name: 'source_package', project: source_project) }
+  let(:broken_packages_backend) do
+    <<~XML
+      <resultlist state="e9e2a9641ea83ac9c8786c24ca57dc6f">
+        <result project="home:foo:Staging:A" repository="openSUSE_Tumbleweed" arch="i586" code="published" state="published">
+          <status package="barbar" code="failed" />
+        </result>
+        <result project="home:foo:Staging:A" repository="openSUSE_Tumbleweed" arch="x86_64" code="published" state="published">
+          <status package="barbar" code="failed" />
+        </result>
+      </resultlist>
+    XML
+  end
+  let(:bs_request) do
+    create(:bs_request_with_submit_action,
+           creator: user,
+           target_project: project.name,
+           target_package: target_package.name,
+           source_project: source_project.name,
+           source_package: source_package.name,
+           description: 'BsRequest 1')
+  end
+
+  let(:broken_packages_path) { "#{CONFIG['source_url']}/build/#{staging_project.name}/_result?code=failed&code=broken&code=unresolvable" }
+
+  describe 'GET #show' do
+    context 'valid user, valid project' do
+      before do
+        login user
+        bs_request.staging_project = staging_project
+        bs_request.save
+        stub_request(:get, broken_packages_path).and_return(body: broken_packages_backend)
+        get :show, params: { name: staging_project.name, format: :xml }
+      end
+
+      it { expect(response).to have_http_status(:success) }
+      it 'returns the staging_project name xml' do
+        assert_select 'staging_project' do
+          assert_select 'staged_requests', 1
+        end
+      end
+    end
```

Comment:
Please add a test case for failures / errors when requesting staging project. And for the case the project doesn't exist.